packages/snap/src/snap.ts

```diff
@@ -22,23 +22,44 @@ const apiDependentMethods = [
 ];
 
 wallet.registerApiRequestHandler(async function (origin: string): Promise<PolkadotApi> {
-  registerOnBalanceChange(wallet, origin);
   return {
     on: (eventName: PolkadotEvent, callback: EventCallback): void => {
```

Comment:
I would rather have more expressive api (not everyone will use this from typescript) so instead of just "on" method, you should expose subscribeToBalance, subscribeToTransactionStatus